Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPIStorageCocoa.mm

```diff
@@ -147,13 +141,20 @@
     }
 
     auto storage = storageForType(storageType);
-    [storage deleteValuesForKeys:createNSArray(keys).get() completionHandler:makeBlockPtr([&, completionHandler = WTFMove(completionHandler)](NSString *errorMessage) mutable {
-        if (errorMessage)
+    [storage getValuesForKeys:createNSArray(keys).get() completionHandler:makeBlockPtr([this, protectedThis = Ref { *this }, keys, storageType, storage, completionHandler = WTFMove(completionHandler)](NSDictionary<NSString *, NSString *> *oldValuesAndKeys, NSString *errorMessage) mutable {
```

Comment:
should this also use `retainPtr`? `storage = RetainPtr { storage }`

Reply:
Yes, or get the storage inside the block again with `storageForType(storageType)`.